### PR TITLE
Refactor EndpointsTreeNode.printTree

### PR DESCRIPTION
**Pull request by nityanandaz**
## :recycle: Current situation
We sometimes use print statements and sometimes use a logger

## :bulb: Proposed solution
Use `app.logger.info` for `EndpointsTreeNode`

## Reviewer Nudging
I can replace more `print`s, if pointed out

**Before**
```
2021-01-21T17:07:42+0100 warning codes.vapor.directory-config : No custom working directory set for this scheme, using /Users/nitai/Library/Developer/Xcode/DerivedData/Apodini-ecnwqazjaoyymagxspghqildiwwa/Build/Products/Debug
/ {
  v1/ {
    - read: Text [0:0]
    greet/ {
      - read: TraditionalGreeter [0:2:-1:0]
    }
    swift/ {
      - read: Text [0:4:-1:0:0]
      5/ {
        3/ {
          - read: Text [0:4:-1:0:1:0]
        }
      }
    }
    user/ {
      :D608F60E-961B-4D24-B176-49C135EF21E4/ {
        - read: UserHandler [0:5:-1:0]
      }
    }
    rand/ {
      - read: Random [0:3:-1:0]
    }
    auction/ {
      - read: Auction [0:1:-1:0]
    }
  }
}
...
```

**After**
```
2021-01-21T17:12:09+0100 warning codes.vapor.directory-config : No custom working directory set for this scheme, using /Users/nitai/Library/Developer/Xcode/DerivedData/Apodini-ecnwqazjaoyymagxspghqildiwwa/Build/Products/Debug
2021-01-21T17:12:09+0100 info org.apodini.application : /
→ v1/
  - read: Text [0:0]
  → auction/
    - read: Auction [0:1:-1:0]
  → swift/
    - read: Text [0:4:-1:0:0]
    → 5/
      → 3/
        - read: Text [0:4:-1:0:1:0]
  → rand/
    - read: Random [0:3:-1:0]
  → user/
    → :A24D4351-C3F1-4F8A-958C-6494A54F0CB1/
      - read: UserHandler [0:5:-1:0]
  → greet/
    - read: TraditionalGreeter [0:2:-1:0]
...
```

### CODE REVIEW

File: Sources/Apodini/Semantic Model Builder/Model/EndpointsTreeNode+CustomDebugStringConvertible.swift [hunk 1]
```diff
@@ -0,0 +1,29 @@
...
+
+extension EndpointsTreeNode: CustomDebugStringConvertible {
+    var debugDescription: String {
+        if let parent = parent {
+            return parent.debugDescription
+        }
+        
+        let node = Node(root: self) { root in
+            Array(root.children)
+        }
```
Comment (bauer-andreas) [inline Sources/Apodini/Semantic Model Builder/Model/EndpointsTreeNode+CustomDebugStringConvertible.swift:13]: Would you mind explaining the usage of Node here? Aren't those currently still two separate things?

Comment (nityanandaz) [reply Sources/Apodini/Semantic Model Builder/Model/EndpointsTreeNode+CustomDebugStringConvertible.swift:13]: We can still leverage the advantage that `Node` is generic. Therefore, we wouldn't need to implement `.description` for another Tree again.

I don't mean to bind `Node` to TypeReflection. How about I move the file to `Apodini/Utilities`?

Comment (lschlesinger) [reply Sources/Apodini/Semantic Model Builder/Model/EndpointsTreeNode+CustomDebugStringConvertible.swift:13]: I think this refactoring is a good idea :) also moving `Node` to `Apodini/Utilities` seems more suitable!

**Review by bauer-andreas — COMMENTED** [review]
Looks good to me. 👍 

Just for the overview, would you mind posting new vs. old output. Would be great for a better overview imo

File: Sources/Apodini/Semantic Model Builder/SharedSemanticModelBuilder.swift [hunk 2]
```diff
@@ -94,10 +94,10 @@ class SharedSemanticModelBuilder: SemanticModelBuilder, InterfaceExporterVisitor
         
         webService.finish()
         
-        webService.root.printTree() // currently only for debugging purposes
+        app.logger.info(.init(stringLiteral: webService.root.debugDescription))
```
Comment (bauer-andreas) [inline Sources/Apodini/Semantic Model Builder/SharedSemanticModelBuilder.swift:97]: Is the init needed? Shouldn't LogMessage be able to be implicitly constructed from a String instance?

**Review by bauer-andreas — APPROVED** [review]
Sorry for the delay from my side. Thanks for tackling this 👍

File: Sources/Apodini/Semantic Model Builder/SharedSemanticModelBuilder.swift (hunk 2 above)
Comment (nityanandaz) [reply Sources/Apodini/Semantic Model Builder/SharedSemanticModelBuilder.swift:97]: Yes, it's needed 😕. As you said, it's a String instance, not a String literal (a value that appears directly in your source code)

**Pull request merged**